x/staking/keeper/delegation.go

```diff
@@ -660,38 +648,32 @@ func (k Keeper) InsertRedelegationQueue(ctx context.Context, red types.Redelegat
 	return k.SetRedelegationQueueTimeSlice(ctx, completionTime, timeSlice)
 }
 
-// RedelegationQueueIterator returns all the redelegation queue timeslices from
-// time 0 until endTime.
-func (k Keeper) RedelegationQueueIterator(ctx context.Context, endTime time.Time) (storetypes.Iterator, error) {
-	store := k.storeService.OpenKVStore(ctx)
-	return store.Iterator(types.RedelegationQueueKey, storetypes.InclusiveEndBytes(types.GetRedelegationTimeKey(endTime)))
-}
-
 // DequeueAllMatureRedelegationQueue returns a concatenated list of all the
 // timeslices inclusively previous to currTime, and deletes the timeslices from
 // the queue.
 func (k Keeper) DequeueAllMatureRedelegationQueue(ctx context.Context, currTime time.Time) (matureRedelegations []types.DVVTriplet, err error) {
-	store := k.storeService.OpenKVStore(ctx)
-
 	// gets an iterator for all timeslices from time 0 until the current Blockheader time
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	redelegationTimesliceIterator, err := k.RedelegationQueueIterator(ctx, sdkCtx.HeaderInfo().Time)
+	redelegationTimesliceIterator, err := k.RedelegationQueue.Iterate(ctx, (&collections.Range[time.Time]{}).EndInclusive(sdkCtx.HeaderInfo().Time))
 	if err != nil {
 		return nil, err
 	}
 	defer redelegationTimesliceIterator.Close()
 
 	for ; redelegationTimesliceIterator.Valid(); redelegationTimesliceIterator.Next() {
-		timeslice := types.DVVTriplets{}
-		value := redelegationTimesliceIterator.Value()
-		if err = k.cdc.Unmarshal(value, &timeslice); err != nil {
-			return nil, err
+		timeslice, err := redelegationTimesliceIterator.Value()
+		if err != nil {
+			return matureRedelegations, err
 		}
 
 		matureRedelegations = append(matureRedelegations, timeslice.Triplets...)
 
-		if err = store.Delete(redelegationTimesliceIterator.Key()); err != nil {
-			return nil, err
+		key, err := redelegationTimesliceIterator.Key()
+		if err != nil {
+			return matureRedelegations, err
+		}
+		if err = k.RedelegationQueue.Remove(ctx, key); err != nil {
+			return matureRedelegations, err
```

Comment:
q: is it safe to do removals like this? In collections the `Clear` method (`deleteDomain`) first stores all the keys in a slice, closes the iterator and then proceeds to iterate through the keys and deletes them.
I think I read somewhere in the docs this wasn't safe but can't find it, asking just in case cc: @testinginprod